frontend/src/screens/study-details.tsx

```diff
@@ -10,13 +10,25 @@ import dayjs from 'dayjs'
 
 const LaunchStudyButton: React.FC<{ study: ParticipantStudy }> = ({ study }) => {
     const api = useStudyApi()
+    const [isBusy, setBusy] = useState(false)
+    const onLaunch = async () => {
+        setBusy(true)
```

Comment:
As far as I can tell this is asynchronous so if the UI is slow enough you can probably still double-click before it disables?

Reply:
well, it's complicated :)   setState is indeed async, BUT react will call it first before re-rendering and accepting any input events.  That has the effect of making sure the button is disabled via the 'busy' prop before any new browser click events